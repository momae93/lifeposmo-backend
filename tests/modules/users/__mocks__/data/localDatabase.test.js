const localDatabase = require('../../../../../src/modules/users/__mocks__/data/localDatabase');

describe('[MODULES][USERS][__MOCKS__][DATAS] : mockData', () => {
  it('should returns an object containing mocked datas', async () => {
    // FUNCTION TESTED
    const mockLocalDatabase = localDatabase;

    // EXPECTED
    const expectedMockLocalDatabase = {
      users: expect.any(Array),
      favoriteUsers: expect.any(Array),
    };

    expect(mockLocalDatabase).toMatchSnapshot();
    expect(mockLocalDatabase).toMatchObject(expectedMockLocalDatabase);
  });
});
