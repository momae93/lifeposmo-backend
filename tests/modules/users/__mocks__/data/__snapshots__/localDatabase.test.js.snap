// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[MODULES][USERS][__MOCKS__][DATAS] : mockData should returns an object containing mocked datas 1`] = `
Object {
  "favoriteUsers": Array [
    Object {
      "id": 1,
      "idUser": 1,
      "idUserFavorite": 2,
    },
  ],
  "users": Array [
    Object {
      "birthdate": 1996-09-27T22:00:00.000Z,
      "description": "Description of Mickael Au",
      "firstname": "Mickael",
      "id": 1,
      "isMale": true,
      "lastname": "AU",
      "password": "NotSoEncryptedPwd",
      "username": "mickaelau",
    },
    Object {
      "birthdate": 1993-03-11T23:00:00.000Z,
      "description": "Description of John Doe",
      "firstname": "John",
      "id": 2,
      "isMale": true,
      "lastname": "DOE",
      "password": "NotSoEncryptedPwd",
      "username": "johndoe",
    },
  ],
}
`;
